mantidimaging/core/io/test/filenames_test.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (C) 2022 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+
+from pathlib import Path
+import unittest
+from unittest import mock
+
+from parameterized import parameterized
+
+from mantidimaging.test_helpers import FileOutputtingTestCase
+from ..filenames import FilenameGroup, FilenamePattern
+
+
+class FilenamePatternTest(unittest.TestCase):
+    @parameterized.expand([
+        ("foo", "foo", 0, ""),
+        ("foo.txt", "foo", 0, ".txt"),
+        ("foo_0001.txt", "foo_", 4, ".txt"),
+        # The 180 here is not an index
+        ("IMAT00006388_PSI_cylinder_Sample_180deg.tif", "IMAT00006388_PSI_cylinder_Sample_180deg", 0, ".tif"),
+        ("IMAT00006388_PSI_cylinder_Sample_000.tif", "IMAT00006388_PSI_cylinder_Sample_", 3, ".tif"),
+        ("IMAT_Flower_Dark_Before_000000.tif", "IMAT_Flower_Dark_Before_", 6, ".tif"),
+        ("IMAT_Flower_Dark_Before.json", "IMAT_Flower_Dark_Before", 0, ".json"),
+    ])
+    def test_pattern_from_name(self, filename, prefix, count, suffix):
+        p1 = FilenamePattern.from_name(filename)
+        self.assertEqual(p1.prefix, prefix)
+        self.assertEqual(p1.digit_count, count)
+        self.assertEqual(p1.suffix, suffix)
+
+    def test_pattern_generate(self):
+        p1 = FilenamePattern.from_name("IMAT_Flower_Tomo_000007.tif")
+        self.assertEqual(p1.generate(23), "IMAT_Flower_Tomo_000023.tif")
+        self.assertEqual(p1.generate(0), "IMAT_Flower_Tomo_000000.tif")
+
+    def test_pattern_match(self):
+        p1 = FilenamePattern.from_name("IMAT_Flower_Tomo_000007.tif")
+        self.assertTrue(p1.match("IMAT_Flower_Tomo_000023.tif"))
+        self.assertFalse(p1.match("IMAT_Flower_Tomo_0000.tif"))
+        self.assertFalse(p1.match("IMAT_Flower_Tomo.tif"))
+        self.assertFalse(p1.match("IMAT_Flower_Tomo.json"))
+        self.assertFalse(p1.match("IMAT_Flower_Tomo_000007.tiff"))
+
+    def test_pattern_dont_match_180(self):
+        p1 = FilenamePattern.from_name("IMAT00006388_PSI_cylinder_Sample_000.tif")
+        self.assertTrue(p1.match("IMAT00006388_PSI_cylinder_Sample_943.tif"))
+        self.assertFalse(p1.match("IMAT00006388_PSI_cylinder_Sample_180deg.tif"))
+        self.assertFalse(p1.match("IMAT00006388_PSI_cylinder_Sample_360deg.tif"))
+
+    def test_pattern_match_different_digits(self):
+        # Allow cases where index has grown above padding
+        p1 = FilenamePattern.from_name("img_000.tif")
+        self.assertTrue(p1.match("img_000.tif"))
+        self.assertTrue(p1.match("img_001.tif"))
+        self.assertTrue(p1.match("img_999.tif"))
+        self.assertTrue(p1.match("img_1000.tif"))
+        self.assertFalse(p1.match("img_0000.tif"))
+
+    def test_pattern_get_value(self):
```

Comment:
Thanks for renaming those bits. I think this unit test needs updating to `test_pattern_get_index` and then everything looks good to go in.